layers/+lang/haskell/packages.el

```diff
@@ -138,6 +138,8 @@
     :defer t
     :config
     (progn
+      (add-hook 'dante-mode-hook
```

Comment:
I think this is not the right spot, you are using functions from flycheck though it could be that flycheck is not installed at all. The only safe place for this is in the init function of a flycheck specific extension or a post-init-flycheck block.

Here is an example from the go layer:
```elisp
(defun go/post-init-flycheck ()
   ;;; Do your specific activation code, runs once during init
  (spacemacs/enable-flycheck 'go-mode) )
```
Btw, I have noticed you would call haskell-hlint only if only "info" level messages from the "haskell-dante" are available. Is that what you intended? I would expect that hlint is called when the error levels drop to "warnings" instead.

Reply:
Hm, unfortunately I don't have much experience with elisp, nor with how elisp code is organized, so I find it difficult to figure out how to organize things correctly. But I hope this is an opportunity to learn more.

Ok, I moved the code into haskell/post-init-flycheck for now. I guess other location could have been init-flycheck-haskell? How do you decide where to put this, what is the difference?

This is probably caused by my inexperience with elisp, but I can't figure out where are haskell/post-init-flycheck and haskell/init-flycheck-haskell used? I tried searching project-wide for them but I couldn't find any usage. On the other hand I noticed other layers have similar functions. So I am guessing they are defined by some naming convention and then some other part of spacemacs pick them up? When in situation like this, how do you go about figuring this stuff out, should I be looking into spacemacs docs, flycheck docs, where do I search?

Regarding "info" -> makes sense, it also seems more reasonable to me to use "warning" -> warnings means it could be ignored, and we want linting to work even if there is warning from compiler that we are ignoring. I used "info" because I copied that line of code from Dante repo README https://github.com/jyp/dante. I wonder why they used it though.

Reply:
As you already have guessed these post-*-config functions are handlers called during the loading process. These are only called if the respective layer is loaded.

The init function for flycheck just registeres a new checker the layer has still to activate flycheck for its modes if it makes sense.

Here is a handy document describing the loading process:
https://github.com/syl20bnr/spacemacs/blob/develop/doc/LAYERS.org

As to why you cannot find them this is due to the generic nature of lisp these functions are called dynamically during the loading process in the spacemacs core.

Reply:
Awesome, I see everything is documented here! I will make sure to check this before proceeding, so I have better understanding of how layers work. Thanks!